### ISSUE TITLE
🐛 `useUniqueElementIds` should ignore SVG elements

### ISSUE THREAD

**Issue by TheKnarf** [biomejs/biome#6206]
> The id global attribute defines an identifier (ID) which must be unique in the whole document.

- [MDN id](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Global_attributes/id)

The point of `useUniqueElementIds` is to ensure that React developers doesn't create static hard coded id's, since React components are reusable and can't statically ensure that id's are unique. Instead one should use reacts built inn `useId`-hook.

However this only applies to HTML! SVG apparently also have an `id`-prop that does not need to be unique across the whole page, as it only refer to elements inside the SVG element. The linting rule needs to be updated to ignore SVG elements.

Example legal SVG:

```html
<svg className="absolute inset-0 size-full bg-white stroke-steel-900/10" fill="none">
   <defs>
        <!--                   \/ local internal svg id defined here         --!>
        <pattern id="pattern-41542215-11f8-4ef6-8c7c-b6dad8789af9" x="0" y="0" width="10" height="10" patternUnits="userSpaceOnUse">
        <path d="M-3 13 15-5M-5 5l18-18M-1 21 17 3"></path>
     </pattern>
   </defs>
   <!--                   the same id is refered to here \/      --!>
   <rect stroke="none" fill="url(#pattern-41542215-11f8-4ef6-8c7c-b6dad8789af9)" width="100%" height="100%"></rect>
</svg>
```

[First reported by @ivan-kleshnin](https://github.com/biomejs/biome/discussions/6025#discussioncomment-13350920).

### Environment information

Biome v2 beta

### What happened?

1. The linting rule triggers on SVG elements in React

### Expected result

The linting rule should ignore SVG elements

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by Tommytrg** [comment]
I'd like to start working on this. I believe most of it can be based on the work done in the PR #6213

**Comment by chengyixu** [comment]
Hi! I'd like to tackle this.

**Approach:** The `useUniqueElementIds` rule currently fires on all elements with static `id` attributes, including SVG elements. I'll update the rule logic in Rust/TypeScript to check if the element is within an SVG context (i.e., the element itself or an ancestor is `<svg>`) and skip the diagnostic in that case, since SVG `id` attributes serve a different purpose (e.g., `<use href="#icon">`) and the uniqueness concern doesn't apply to reusable SVG symbols.
**Timeline:** 2-3 days
**Background:** Node.js/TypeScript developer, 35+ npm packages published. Familiar with AST traversal and linting rule development.

Ready to start immediately. — @chengyixu

**Comment by dyc3** [comment]
@chengyixu hey, mr robot, i know you have good intentions, but there are literally 2 open PRs for this already, and someone has already been assigned to the issue. please make a little more effort to avoid duplicating work.

**Comment by ematipico** [comment]
Closing. The issues has been fixed in one of latest versions of Biome.
